src/cdi/CDI.cc

```diff
@@ -211,36 +211,43 @@ void CDI::integrate_Rosseland_Planckian_Spectrum(const size_t groupIndex, const
  */
 void CDI::integrate_Planckian_Spectrum(std::vector<double> const &bounds, double const T,
                                        std::vector<double> &planck) {
-  Require(T >= std::numeric_limits<double>::min() && T < std::numeric_limits<double>::max());
-  Require(bounds.size() > 0);
+  Require(bounds.size() > 1);
+  Require(bounds[0] >= 0.0);
+  // Subsequent bounds must be monotonicallly increasing; checked in loop below
+  Require(T >= 0.0);
 
   size_t const groups(bounds.size() - 1);
   planck.resize(groups, 0.0);
 
-  // nu/T must be < numeric_limits<double>::max(), throw assertion in DBC mode if this occurs
-  Check(T > bounds.back() * std::numeric_limits<double>::min());
-
-  // Initialize the loop:
-  double scaled_frequency = bounds[0] / T;
-  double planck_value = extend ? 0.0 : integrate_planck(scaled_frequency);
-
-  for (size_t group = 0; group < groups; ++group) {
-    // Shift the data down:
-    Remember(double const last_scaled_frequency = scaled_frequency;);
-    double const last_planck = planck_value;
-
-    // New values:
-    scaled_frequency = bounds[group + 1] / T;
-    Ensure(scaled_frequency > last_scaled_frequency);
-    planck_value = integrate_planck(scaled_frequency);
-
-    // Record the definite integral between frequencies.
-    planck[group] = planck_value - last_planck;
-    Ensure(planck[group] >= 0.0);
-    Ensure(planck[group] <= 1.0);
-  }
-  if (extend) {
-    planck[groups - 1] += 1 - planck_value;
+  if (T > 0.0) {
+    // Initialize the loop:
+    double frequency = bounds[0];
+    double planck_value =
+        extend ? 0.0 : ((frequency < T * 50) ? integrate_planck(frequency / T) : 1.0);
```

Comment:
`frequency / T` can overflow if `T` is too small.  Can you re-incorporate something akin to this line that you removed?
```c++
  // nu/T must be < numeric_limits<double>::max(), throw assertion in DBC mode if this occurs
  Check(T > bounds.back() * std::numeric_limits<double>::min());
```
One possibility would be changing the conditional `if(T>0.0)` to `if(T>bounds[0]*std::numeric_limits<double>::min())` and adjusting the `else` portion accordingly for `T=epsilon`.

A similar piece of logic will be required below where you call `integrate_planck(frequency / T)`

Reply:
The magic number `50` should be documented.  I don't understand why this value appears here (and below).

Reply:
These two go together.  If T is zero, there's a special branch to handle that. If T is nonzero, frequency< T * 50 guarantees the division is well-behaved and cannot overflow. The magic value T*50 is the point on the upper tail where the integral becomes 1 to roundoff. (Actually, the value is around 48; choosing 50 gives just a little more slack if a compiler does something weird and wonderful with its optimizations.)

But, yeah, it seems clear I need to document that.